component/src/main/wso2/kafka/producer.bal

```diff
@@ -79,82 +79,83 @@ public type ProducerConfig record {
     boolean enableIdempotence = false;          // ENABLE_IDEMPOTENCE_CONFIG 0
 };
 
-documentation { Represent a Kafka producer endpoint
-    E{{}}
-    F{{producerActions}} handle all the actions related to the endpoint
-    F{{producerConfig}} used to store configurations related to a Kafka connection
-}
+# Represent a Kafka producer endpoint.
+#
+# + producerActions - handle all the actions related to the endpoint
+# + producerConfig - used to store configurations related to a Kafka connection
 public type SimpleProducer object {
 
     public ProducerAction producerActions;
     public ProducerConfig producerConfig;
 
-    documentation { Initialize the producer endpoint
-        P{{config}} configurations related to the endpoint
-    }
+    # Initialize the producer endpoint.
+    #
+    # + config - configurations related to the endpoint.
     public function init(ProducerConfig config) {
         self.producerConfig = config;
         self.producerActions.init(config);
     }
 
-    documentation { Registers producer endpoint in the service
-        P{{serviceType}} type descriptor of the service
-    }
+    # Registers producer endpoint in the service.
+    #
+    # + serviceType - type descriptor of the service.
     public function register (typedesc serviceType) {
 
     }
 
-    documentation { Starts the consumer endpoint }
+    # Starts the consumer endpoint.
     public function start () {}
 
-    documentation { Returns the action object of ProducerAction }
-    public function getCallerActions () returns (ProducerAction) {
+    # Returns the action object of ProducerAction.
+    #
+    # + returns - Producer actions.
+    public function getCallerActions () returns ProducerAction {
         return producerActions;
     }
 
-    documentation { Stops the consumer endpoint }
+    # Stops the consumer endpoint.
     public function stop () {
         self.producerActions.close();
     }
 
 };
 
-documentation { Kafka producer action handling object }
+# Kafka producer action handling object.
+#
+# + producerHolder -
+# + connectorID -
 public type ProducerAction object {
 
     public map producerHolder;
     public string connectorID = system:uuid();
 
-    documentation { value:"Simple Send action which produce records to Kafka server
-        P{{value}} record contents
-        P{{topic}} topic the record will be appended to
-        P{{key}} key that will be included in the record
-        P{{partition}} partition to which the record should be sent
-        P{{timestamp}} timestamp of the record, in milliseconds since epoch
-    }
+    # Simple Send action which produce records to Kafka server.
+    #
+    # + value - record contents.
+    # + topic - topic the record will be appended to.
```

Comment:
Better if we state this as `topic - topic to which the record will be appended.

Reply:
Fixed in 8e52317